sulaimonlasisi/clearance-app
# Integrate Amazon api and refactor walmart code

### Issue
https://github.com/sulaimonlasisi/clearance-app/issues/3

### What was changed?
1) Integrated the amazon api with a node js client into the project.
You will need to add the following to your `.env` in order to run the amazon services:
`AMAZON_ACCESS_KEY=`
`AMAZON_SECRET_KEY=`
`AMAZON_SELLER_ID=`
ask me if you need the values.
2) Heavy refactoring to existing walmart operations.
Code is much cleaner and `getSpecialFeedsItems` is significantly faster. `getPaginatedSpecialFeeds` is slightly faster but will still take a long time to complete.
3) Can now run app with `npm start`. See README for details.

### How to test?
1) `npm install` to pull down new dependencies.
2) Run `testAmazonProducts`, `getSpecialFeedsItems`,  `getPaginatedSpecialFeeds` from `simple.js`. I recommend only running one at a time and commenting out the others. The amazon function is just using dummy data at the moment for proof of concept. See README for instructions on running project.

Ignore all of the `/node_modules` changes in this PR.
